src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -139,8 +139,8 @@ SELECT COUNT(c) > 0
     @EntityGraph(type = LOAD, attributePaths = { "lectures", "lectures.attachments" })
     Optional<Course> findWithEagerLecturesById(long courseId);
 
-    @EntityGraph(type = LOAD, attributePaths = { "exercises", "lectures", "lectures.attachments" })
-    Optional<Course> findWithEagerExercisesAndLecturesById(long courseId);
+    @EntityGraph(type = LOAD, attributePaths = { "exercises", "exercises.plagiarismDetectionConfig", "exercises.teamAssignmentConfig", "lectures", "lectures.attachments" })
+    Optional<Course> findWithEagerExercisesAndExerciseDetailsAndLecturesById(long courseId);
```

Comment:
Maybe one could be more precise in the naming when fetching the details instead of `...AndExerciseDetails`

Reply:
Sure, but honestly, it doesn't really add anything but length. If you want to know what exercise details are included, you can check it out.

Do you consider it critical to include the precise terms?

Reply:
It is not critical that's true I just thought, it does increase the length but when you call the method one would always know what you get without digging deeper. But except that code looks good to me. If you want to keep it that way, just let me know and I'll approve.